llvm/lib/Transforms/Scalar/InductiveRangeCheckElimination.cpp

```diff
@@ -985,13 +981,25 @@ bool InductiveRangeCheckElimination::run(
   }
 
   LLVMContext &Context = Preheader->getContext();
+
+  const char *FailureReason = nullptr;
```

Comment:
Can we extract moving LoopStructure::parseLoopStructure into a separate NFC patch?